bam_cat.c

```diff
@@ -604,13 +648,13 @@ int main_cat(int argc, char *argv[])
     switch (hts_get_format(in)->format) {
```

Comment:
The output file format is still decided by the format of the first input file, which makes the `--output-fmt` option useless.

Reply:
Done